### PR TITLE
Makman2/linterpack2

### PR DESCRIPTION
**Pull request by Makman2**

### CODE REVIEW

File: bears/java/InferBear.py [hunk 1]
```diff
@@ -1,25 +1,16 @@
...
-    severity_map = {
-        "error": RESULT_SEVERITY.MAJOR,
-        "warning": RESULT_SEVERITY.NORMAL}
+@linter(executable='infer',
+        output_format='regex',
+        output_regex=r'.+:(?P<line>\d+): (?P<severity>error|warning): '
+                     r'(?P<message>.*)')
+class InferBear:
+    """
+    Checks the code with ``infer``.
```
Comment (SanketDG) [inline bears/java/InferBear.py]: more info needed.

File: bears/js/JSComplexityBear.py [hunk 2]
```diff
@@ -1,36 +1,32 @@
...
+from coalib.bearlib.abstractions.Linter import linter
 from coalib.results.Result import Result
 
 
-class JSComplexityBear(LocalBear, Lint):
-    executable = "cr"
+@linter(executable='cr')
+class JSComplexityBear:
+    """
+    Calculates cyclomatic complexity using ``cr``.
```
Comment (SanketDG) [inline bears/js/JSComplexityBear.py]: same here.

Comment (SanketDG) [inline bears/js/JSComplexityBear.py]: and all other docstrings.

**Comment by Makman2** [comment]
@SanketDG I've just copied the docs^^ Would be better to have a separate commit for improved docs^^

**Comment by sils1297** [comment]
@Makman2 what's the state here?

**Comment by Makman2** [comment]
CI is failing and I don't know why, but haven't really looked at it either :3

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/394/commits/39c3d5d7b1b45fc725bb74ae59bd89441af34c1a https://github.com/coala-analyzer/coala-bears/pull/394/commits/d1ffd984bae034076244ac4449632a1aa04d5ffe https://github.com/coala-analyzer/coala-bears/pull/394/commits/f0833e9f4cacf9918ffe440f0a574e50df960990 https://github.com/coala-analyzer/coala-bears/pull/394/commits/f68106611156b07019d90dd570340d2770b5ad93 https://github.com/coala-analyzer/coala-bears/pull/394/commits/bb8a4ae175f659bea1d9f901c8744fab7d552919

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**